File: sample_inference_precision.py
#!/usr/bin/env python3
import sys

import openvino as ov
import openvino.properties.hint as hints


def parse_precision(name: str):
    mapping = {
        "bf16": ov.Type.bf16,
        "f16": ov.Type.f16,
        "f32": ov.Type.f32,
    }
    key = name.lower()
    if key not in mapping:
        raise ValueError(f"Unsupported precision: {name} (use bf16|f16|f32)")
    return mapping[key]


def main() -> int:
    model_path = sys.argv[1] if len(sys.argv) > 1 else "model.xml"
    # hint_name is a Python string (str).
    # Common values for this project: "bf16", "f16", "f32".
    # Example run: python3 sample_inference_precision.py /path/to/model.xml bf16
    hint_name = sys.argv[2] if len(sys.argv) > 2 else "bf16"

    core = ov.Core()
    # HIGHLIGHT: Set inference precision hint before model compilation.
    # The value is passed via the hint_name string.
    core.set_property("CPU", {hints.inference_precision: parse_precision(hint_name)})
    model = core.read_model(model_path)
    compiled_model = core.compile_model(model, "CPU")

    current_hint = compiled_model.get_property(hints.inference_precision())
    # HIGHLIGHT: This is the exact log line for inference precision hint.
    print(f"[INFO] INFERENCE_PRECISION_HINT={current_hint}")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: sample_inference_precision.c
#include <stdio.h>

#include <openvino/c/openvino.h>

int main(int argc, char** argv) {
    const char* model_path = (argc > 1) ? argv[1] : "model.xml";
    // infer_precision_hint is a C string (const char*).
    // Common values for this project: "bf16", "f16", "f32".
    // Example run: ./sample_c /path/to/model.xml bf16
    const char* infer_precision_hint = (argc > 2) ? argv[2] : "bf16";

    ov_core_t* core = NULL;
    ov_compiled_model_t* compiled_model = NULL;
    char* current_hint = NULL;

    if (ov_core_create(&core) != OK) {
        fprintf(stderr, "[ERROR] ov_core_create failed\n");
        return 1;
    }

    // HIGHLIGHT: Set inference precision hint before model compilation.
    // The value is passed as a string via infer_precision_hint.
    if (ov_core_set_property(core,
                             "CPU",
                             ov_property_key_hint_inference_precision,
                             infer_precision_hint) != OK) {
        fprintf(stderr, "[ERROR] set INFERENCE_PRECISION_HINT failed: %s\n", infer_precision_hint);
        ov_core_free(core);
        return 1;
    }

    if (ov_core_compile_model_from_file(core,
                                        model_path,
                                        "CPU",
                                        0,
                                        &compiled_model) != OK) {
        fprintf(stderr, "[ERROR] compile model failed: %s\n", model_path);
        ov_core_free(core);
        return 1;
    }

    if (ov_compiled_model_get_property(compiled_model,
                                       ov_property_key_hint_inference_precision,
                                       &current_hint) == OK &&
        current_hint != NULL) {
        // HIGHLIGHT: This is the exact log line for inference precision hint.
        printf("[INFO] INFERENCE_PRECISION_HINT=%s\n", current_hint);
        ov_free(current_hint);
    } else {
        fprintf(stderr, "[WARN] failed to read INFERENCE_PRECISION_HINT\n");
    }

    ov_compiled_model_free(compiled_model);
    ov_core_free(core);
    return 0;
}
